Manual.md

```diff
@@ -0,0 +1,145 @@
+
+- `self.prebuild` - Can be used to define functions<sup>(Or commands)</sup> that should be used to edit various files.
+
+- `def self.patch` - Can be used to define functions<sup>(Or commands)</sup> that should be used to `patch` various files.
+- `def self.preinstall` - Can be used to define functions<sup>(Or commands)</sup> that should happen `pre`-install. 
+- `def self.check` - Can be used to define functions<sup>(Or commands)</sup> that should be used to check the compile binary.
+- `def self.postinstall` - Can be used to define functions<sup>(Or commands)</sup> that should happen `post`-install.
+***
+# Rundown
+
+The rundown of what each function and variable are/(can be) used for follows.
+
+```ruby
+require 'package' # must occour within each `.rb`
+			      # Notice the newline
+class Template < Package # Notice the capitals, EG: 'I3' - would be used for an 'i3' package
+  description 'The template script' # Notice the indent, should contain no more than one line of text
+  homepage '#' # Notice the same indent, EG: 'https://i3wm.org/' - Would be used for an 'i3' Package
+  version 'version#-revision#' # EG: 4.18.2-1 - Where '4.18.2' is the version and '1' is the script revision
+  compatibility '#' # Can contain 'all', or a list of supported architectures each seperated by a space
+  source_url '#' # The URL from which the source of the package can be downloaded
+  source_sha256 '#' # The `sha256` checksum of the downloaded source pcakage
+                  # Notice the newline
+  depends_on '#' # Soft where this package depends on
+  depends_on '#' => :build # Only required when the package is built from source
+  				  # Notice the newline
+  self.prebuild # For sed operations, not required
+    system '#' # Replace '#' with a sed command
+  end
+  self.patch # For patch operations, not required
+    system '#' # Replace '#' with a patch command
+  end
+				  # Notice the newline
+  self.build # Contains the commands which compile the package, required if package con be compiled
+    system '#' # Replace '#' with commands to compile the package 
+    system '#' # Should contain something like, "./configure --prefix=/usr/local"
+    system '#' # Use incase additions commands are required
+    system '#' # Should contain something like, "'make -j$(nproc)'"
+  end
+				  # Notice the newline
+  self.preinstall # For pre-install conditions, not required
+    system '#'
+  end
+				  # Notice the newline
+  self.install # For make install, required if package can be compiled
+    system '#' # Should contain something like, "'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'"
+  end
+				  # Notice the newline
+  self.postinstall # For post-install messages or operations, not required
+    system '#' 
+  end
+end
+```
+***
+# Footnotes
+
+<a name="Arch">**Arch**</a>: Chromebrew packages for `x86_64`, `i686`, `armv7l`, and `aarch64`, try and make your package work on all of these architectures.  
```

Comment:
This is a lot to ask of package maintainers.  Most users probably don't have 4 different Chromebooks for each architecture.